### PR TITLE
feat(transform): add support for loading and joining an array of vrl files in remap

### LINKED ISSUE #21485: Feature Request - Add files property to remap transform
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Use Cases

The proposed feature would allow users to break down Vector remaps into smaller, more manageable units of configuration. This would enable:

1. Improved organization of VRL code
2. Easier sharing of VRL snippets across components
3. Better testability of individual remap sections
4. Simplified maintenance of complex transformations

### Attempted Solutions

Currently, users can only specify a single file for the `remap` transform using the `file` property. To achieve similar functionality, users might:

1. Create multiple `remap` components, which can be cumbersome and less efficient.
2. Use external tools like jsonnet to combine multiple VRL files, adding complexity to the configuration process.

However, these solutions do not provide the seamless experience that a native `files` property would offer.

### Proposal

Add a new `files` property to the `remap` transform that accepts an array of file paths. When `files` is provided (and `file` or `source` are not), Vector would concatenate the contents of the specified VRL files with a newline separator.

Example configuration:

```yaml
transforms:
  my_transform_id:
    type: remap
    inputs:
      - my-source-or-transform-id
    files:
      - ./my/remap_a.vrl
      - ./my/remap_b.vrl
```

This would be equivalent to:

```yaml
transforms:
  my_transform_id:
    type: remap
    inputs:
      - my-source-or-transform-id
    source: |
      # Contents of ./my/remap_a.vrl
      . = parse_json!(string!(.message))
      .timestamp = to_unix_timestamp(to_timestamp!(.timestamp))

      # Contents of ./my/remap_b.vrl
      del(.username)
      .message = downcase(string!(.message))
```

This feature would allow users to organize their VRL code into smaller, more manageable files while still benefiting from the efficiency of a single `remap` transform.

### References

No directly related issues found. This feature request is based on user needs for improved VRL management and configuration flexibility.

- [Conversation about the feature in Discord](https://discord.com/channels/742820443487993987/746076283074773150/1294115755843129435)

### Version

0.41.0

### PR DESCRIPTION
**Pull request by brittonhayes**
# Description

This PR adds support for loading and concatenating multiple VRL files in the remap transform. This new feature allows users to break down Vector remaps into smaller, more manageable units of configuration, improving organization, reusability, and maintainability of VRL code.

## Changes

- Added a new `files` property to the `RemapConfig` struct that accepts an `Option<Vec<PathBuf>>`.
- Updated the `compile_vrl_program` method to handle the new `files` option:
  - When `files` is specified, the contents of all specified VRL files are read and concatenated with newline separators.
  - The combined content is then used as the VRL program source.
- Modified the error handling to include the new `files` option in the `SourceAndOrFileOrFiles` error variant.
- Added a new `read_file` helper method to handle file reading operations.

## Use Cases

This feature enables:

1. Improved organization of VRL code
2. Easier sharing of VRL snippets across components
3. Better testability of individual remap sections
4. Simplified maintenance of complex transformations

## Example Usage

```yaml
transforms:
  my_transform_id:
    type: remap
    inputs:
      - my-source-or-transform-id
    files:
      - ./my/remap_a.vrl
      - ./my/remap_b.vrl
```

```yaml
transforms:
  my_transform_id:
    type: remap
    inputs:
      - my-source-or-transform-id
    source: |
      # Contents of ./my/remap_a.vrl
      . = parse_json!(string!(.message))
      .timestamp = to_unix_timestamp(to_timestamp!(.timestamp))

      # Contents of ./my/remap_b.vrl
      del(.username)
      .message = downcase(string!(.message))
```

This configuration will load and concatenate the contents of `remap_a.vrl` and `remap_b.vrl`, applying them as a single VRL program in the specified order.

## Notes

- The `files` option is mutually exclusive with `source` and `file`. An error will be raised if more than one of these options is specified.
- File paths can be relative to the current working directory or absolute.
- The order of files in the array determines the order of concatenation and execution.

Closes #21485

### CODE REVIEW

**Review by jszwedko — COMMENTED** [review]
Thanks @brittonhayes !

The implementation looks good. Could I ask for a couple of more things:

- Run `make generate-component-docs` to regenerate the documentation with the new field
- Add a changelog fragment per https://github.com/vectordotdev/vector/blob/master/changelog.d/README.md

**Review by jszwedko — APPROVED** [review]
Awesome, thanks @brittonhayes !

**Review by buraizu — APPROVED** [review]

File: changelog.d/21485_load_multiple_vrl_files_in_remap.feature.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Adds support for loading and concatenating multiple VRL files in the remap transform. This allows users to break down Vector remaps into smaller, more manageable units of configuration, improving organization, reusability, and maintainability of VRL code.
```
Comment (pront) [inline changelog.d/21485_load_multiple_vrl_files_in_remap.feature.md:1]: nit: 
```suggestion
Adds support for loading and concatenating multiple VRL files in the `remap` transform. This allows users to break down Vector remaps into smaller, more manageable units of configuration, improving organization, reusability, and maintainability of VRL code.
```

**Comment by britton-from-notion** [comment]
@jszwedko could I bug ya for another test run, I know pavlos is after work hours rn 🙏

**Pull request merged**